File: apps/server/src/api/components/health-check/health-check-service.ts
import { INTERNAL_SERVER_ERROR } from 'http-status'

import { DBService } from '../../../services'
import { ServerError } from '../../server-error'

interface IHealthCheckService {
  checkHealth(): boolean
}

class HealthCheckService implements IHealthCheckService {
  constructor(private dbService: DBService) {}

  checkHealth(): boolean {
    try {
      return this.dbService.isDatabaseAlive()
    } catch (error) {
      const message = 'An error occurred when checking if application is alive'
      throw new ServerError(message, INTERNAL_SERVER_ERROR, {
        context: 'unknown',
        cause: error,
      })
    }
  }
}

export { HealthCheckService, IHealthCheckService }

File: apps/server2/src/api/components/health/health_check_service.py
from abc import ABC, abstractmethod

from fastapi import status
from src.server_error import ServerError
from src.services.db_service import DBService


class IHealthCheckService(ABC):
    @abstractmethod
    async def check_health(self) -> bool:
        raise Exception("NotImplementedException")


class HealthCheckService(IHealthCheckService):
    def __init__(self, db_service: DBService):
        self.db_service = db_service

    async def check_health(self):
        try:
            await self.db_service.migrate_database()
            return await self.db_service.check_database_is_alive()
        except Exception:
            message = "An error occurred when checking if application is alive"
            print(message)
            raise ServerError(
                message,
                status.HTTP_500_INTERNAL_SERVER_ERROR,
                # type(
                #     "Detail",
                #     (object,),
                #     {"context": "unknown", "cause": error},
                # ),
            )
